Add more ido-like functionality

savuori opened the pull request:
The following changes were made to make Closed work more similarly to the ido-mode 
- initialize the filter with the directory of the currently open file or empty string if no file open
  - assign backspace to the new command Delete Filename
  - add new command Delete Filename which deletes the last part of the filter's path

Commit:
Add more ido-like functionality
 * initialize the filter with the path of the currently open file or empty string if no file open
 * assign backspace to the new command Delete Filename
 * add new command Delete Filename which deletes the last part of the filter's path

Commit:
Refactor deleteFilename, initialize filter with the current directory
 * split deleteFilename into deleteFilename (utility function) and deleteFilenameCommand (used as the editor command)

Raynes commented on lib/closed-view.coffee:
```diff
@@ -69,6 +69,21 @@ class ClosedView extends SelectListView
     @panel.hide()
 
+  # Remove the last level, file or directory, from the filter path
+  deleteFilename: (path) ->
```
https://nodejs.org/api/path.html#path_path_dirname_p

Raynes commented on lib/closed-view.coffee:
```diff
@@ -81,6 +96,9 @@ class ClosedView extends SelectListView
     @panel ?= atom.workspace.addModalPanel(item: this)
     @panel.show()
+    currentPath = atom.workspace.getActiveTextEditor()?.buffer?.file?.path;
+    if(currentPath?)
```
Parentheses are unnecessary for if.

Commit:
Use path-module to handle paths instead of homegrown solutions
Use user's home directory (expanded tilde) as the default filter if no files are open

sampovuori commented:
Thanks for the comments. This ought to be cleaner and OS-independent.
Also, made user's home directory the default if no files are open.

Commit:
Remove the pesky parentheses

Raynes commented:
Nice! I love it. Thanks for the contributions, dood. I'll release this.
